audb/core/utils.py

```diff
@@ -106,3 +108,15 @@ def _lookup(
         f'for database '
         f"'{name}'."
     )
+
+
+def _mkdir_tree(
+        files: typing.Sequence[str],
+        root: str,
+):
+    r"""Helper function to create folder tree."""
+    folders = set()
+    for file in files:
+        folders.add(os.path.dirname(file))
```

Comment:
Do you think it is faster as

```python
    folders = set([os.path.dirname(file) for file in files])
```
because we don't need to create a big list first?

Reply:
Currently we check for every file if the folder exists. If you have 2M files, you have that many accesses to the file system. Now, if all 2M files are in the same folder, we only have a single call to the file-system.

Reply:
Sorry, misunderstood your question. I don't think it makes a bit difference if you do it in a for loop or in single line.

Reply:
If you prefer I can change to your suggestion.

Reply:
I tested it on  1,461,007 files, and the result is identical (averaged over 100 repetitions):
* my proposal: 1.155 s
* your proposal: 1.156 s

So, let's stick with your solution.